gpii/node_modules/lifecycleManager/src/UserLogonStateChange.js

```diff
@@ -153,22 +153,32 @@ gpii.lifecycleManager.userLogonHandling.loginUser = function (that, lifecycleMan
 };
 
 gpii.lifecycleManager.getDeviceContext = function (deviceReporter, event, errorEvent) {
+    gpii.lifecycleManager.getDeviceContextPromise(deviceReporter).then(
+        event.fire,
+        errorEvent.fire
+    );
+};
+
+gpii.lifecycleManager.getDeviceContextPromise = function (deviceReporter) {
```

Comment:
Please could you add a comment here which says something like: 
````
// TODO: We have a factoring problem here since properly the LifecycleManager has no business
// reaching out into the environment to interact with the DeviceReporter, as shown by the
// awkward-looking reference {flowManager}.deviceReporter which is out of place with the others.
// However, this ended up being a natural place for the implementation since it is scoped to the 
// matchmaking workflow which is tied to lifecycleManager's userLogonRequest via containment.
// When we improve this factoring we will better deliver on our goal of "containment without dependency"
````

Reply:
You can reference https://issues.gpii.net/browse/GPII-4324

Reply:
commented and referenced.